bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3269,6 +3269,8 @@ void ReplicaImp::addTimers() {
 
 void ReplicaImp::start() {
   ReplicaForStateTransfer::start();
+  if (controlStateManager_ && controlStateManager_->getCheckpointToStopAt().has_value())
```

Comment:
I would add a comment here stating that this is only safe if the system doesn't crash and restart on its own, but is done purposefully during a manual upgrade.

Reply:
done